src/sbvr-api/permissions.ts

```diff
@@ -1213,27 +1214,35 @@ const $getUserPermissions = (() => {
 	);
 	return env.createCache(
 		'userPermissions',
-		async (userId: number) => {
-			const permissions = (await getUserPermissionsQuery()({
-				userId,
-			})) as Array<{ name: string }>;
+		async (userId: number, tx?: Tx) => {
+			const permissions = (await getUserPermissionsQuery()(
+				{
+					userId,
+				},
+				undefined,
+				{ tx },
+			)) as Array<{ name: string }>;
 			return permissions.map((permission) => permission.name);
 		},
 		{
 			primitive: true,
 			promise: true,
+			normalizer: ([userId]: [number]) => `${userId}`,
 		},
 	);
 })();
-export const getUserPermissions = async (userId: number): Promise<string[]> => {
+export const getUserPermissions = async (
+	userId: number,
+	tx?: Tx,
```

Comment:
Does it make sense for anywhere within pinejs to also be passing this in?

Reply:
@Page- the only places this is called inside pine is from `getGuestPermissions()` (used by `getReqPermissions(req, odataBinds)`) and `checkPassword()` (used by the `login()` middleware).
None of the two callers have an active transaction or do a write before those calls, so it doesn't seem necessary to create a read transaction just to pass it in.